### DISCUSSION TITLE
How can I remove the ProductFeatures tab from the "WixUI_Advanced" Installation widget.

### DISCUSSION THREAD

**Discussion by HarshSuman82** [wixtoolset/issues#6490] [Questions] [answered]
I am using "WixUI_Advanced" UI for the installation process but I do not want ProductFeatures tab in installation widget 

![image](https://user-images.githubusercontent.com/86040662/123416776-9e446e80-d5d4-11eb-836e-53fdb821e4f1.png)

**Comment by BMurri** [comment, marked as answer]
Please see [this](https://wixtoolset.org//documentation/manual/v3/wixui/wixui_customizations.html) page.

**Reply by HarshSuman82** [reply]
Thanks, BMurri  I followed the step which is mention in the document I paste all the content which is under <Fragment/> of WixUI_Advanced.wxs under <UI> tag but I am getting duplicate build errors for all the dialogRef.  I have attached a screenshot of the code.  

![image](https://user-images.githubusercontent.com/86040662/124700847-1b43e200-df0b-11eb-82e4-bf809890e8fb.png)


![image](https://user-images.githubusercontent.com/86040662/124700781-fea7aa00-df0a-11eb-809d-33ebdbf86412.png)

**Reply by BMurri** [reply]
### TL;DR: This is why I always suggest using Bundles for all UI, NOT MSI.

Two considerations:

- when copying code/authoring, make sure you are not also referencing the original. For example, you've placed the copy into its own `Fragment`, change the referenced Id of the `UI` tag in both the copy and your reference to it. If you instead copied to under your `Product` tag, then remove the `UIRef` to the original.
- Per the instructions I linked, you also need to change all references to the dialog you are removing from the sequence to the corresponding dialog that the removed dialog was itself referencing. In your example, every `Value="FeaturesDlg"` attribute should update it's value to whatever the most equivalent `Control` would be in the removed dialog (via `Publish` or directly defined).

In this particular case, this consideration means means that: 

- `MaintenanceTypeDlg`'s `ChangeButton` control
- `InstallDirDlg`'s `Next` control
- `AdvancedWelcomeEulaDlg`'s `Next` control
- `InstallScopeDlg`'s `Next` control
All become:

```xml
<Publish Dialog="-->dlgId<--" Control="-->ctlId<--" Event="SpawnWaitDialog" Value="WaitForCostingDlg">NOT Installed AND (!(wix.WixUICostingPopupOptOut) OR CostingComplete = 1)</Publish>
<Publish Dialog="-->dlgId<--" Control="-->ctlId<--" Event="EndDialog" Value="Return"><![CDATA[OutOfDiskSpace <> 1]]></Publish>
<Publish Dialog="-->dlgId<--" Control="-->ctlId<--" Event="SpawnDialog" Value="OutOfRbDiskDlg">OutOfDiskSpace = 1 AND OutOfNoRbDiskSpace = 0 AND (PROMPTROLLBACKCOST="P" OR NOT PROMPTROLLBACKCOST)</Publish>
<Publish Dialog="-->dlgId<--" Control="-->ctlId<--" Event="EndDialog" Value="Return">OutOfDiskSpace = 1 AND OutOfNoRbDiskSpace = 0 AND PROMPTROLLBACKCOST="D"</Publish>
<Publish Dialog="-->dlgId<--" Control="-->ctlId<--" Event="EnableRollback" Value="False">OutOfDiskSpace = 1 AND OutOfNoRbDiskSpace = 0 AND PROMPTROLLBACKCOST="D"</Publish>
<Publish Dialog="-->dlgId<--" Control="-->ctlId<--" Event="SpawnDialog" Value="OutOfDiskDlg">(OutOfDiskSpace = 1 AND OutOfNoRbDiskSpace = 1) OR (OutOfDiskSpace = 1 AND PROMPTROLLBACKCOST="F")</Publish>
```

Hope this helps anyone struggling with this.
